kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/io/kotest/assertions/nondeterministic/EventuallyTest.kt

```diff
@@ -431,6 +431,54 @@ class EventuallyTest : FunSpec() {
             }
          }.message shouldNotContain "The first error was caused by: first"
       }
+
+      test("raise error if duration is less than 0") {
+         val message =
+            shouldThrow<IllegalArgumentException> {
+               eventually((-1).milliseconds) {
+                  1 shouldBe 2
+               }
+            }.message
+
+         message shouldContain "Duration must be greater than or equal to 0"
+      }
+
+      test("raise error if retries is less than 0") {
+         val message =
+            shouldThrow<IllegalArgumentException> {
+               eventuallyConfig {
+                  retries = -1
+               }
+            }.message
+
+         message shouldContain "Retries must be greater than or equal to 0"
+      }
+
+      test("when duration is set to default it cannot end test until iteration is done") {
+         val finalCount = 100
+         var count = 0
+         val config = eventuallyConfig {
+               retries = finalCount
+            }
+         shouldThrow<AssertionError> {
+            eventually(config) {
+               count++
+               1 shouldBe 2
+            }
+         }
+
+         count shouldBe finalCount
+      }
+
+      test("test eventually without configuration") {
+         // linked to issue #3988
+         var cnt = 0
+         eventually {
+            cnt += 1
+            cnt shouldBe 100
+         }
+         cnt shouldBe 100
```

Comment:
```suggestion
         var count = 0
         eventually {
            count += 1
            count shouldBe 100
         }
         count shouldBe 100
```

nitpick; rename var for clarity